Modules/Filtering/ImageGradient/include/itkVectorGradientMagnitudeImageFilter.hxx

```diff
@@ -69,7 +64,7 @@ VectorGradientMagnitudeImageFilter<TInputImage, TRealType, TOutputImage>::PrintS
   }
   os << std::endl;
   os << indent << "m_ComponentWeights = ";
-  for (i = 0; i < VectorDimension; i++)
+  for (i = 0; i < ImageDimension; i++)
```

Comment:
Maybe change to `unsigned int i` here too.

And ImageDimension is not the same than VectorDimension.

Reply:
If you're adjusting this line of code anyway, you may consider the prefix form `++i` instead of the postfix form (`i++`). Sutter & Alexandrescu - _C++ Coding Standards: 101 Rules, Guidelines, and Best Practices_ [28. Prefer the canonical form of ++ and --. Prefer calling the prefix forms](https://www.oreilly.com/library/view/c-coding-standards/0321113586/ch29.html)

Reply:
Thanks both. Yes, definitely those would be welcomed changes. However, those are `STYLE` changes, and in this case I preferred to try to only address the bug in the PR so that it becomes clear where it came from. I will note down these to try to address them in a separate PR once we are sure the bug is resolved. Note that I already meant to do such changes in separate PRs as [noted here](https://github.com/InsightSoftwareConsortium/ITK/issues/2275#issuecomment-770134280).